test/EIP20Gateway/getEIP20BaseToken.js

```diff
@@ -0,0 +1,93 @@
+const chai = require('chai');
+const Web3 = require('web3');
+const sinon = require('sinon');
+const EIP20Gateway = require('../../src/ContractInteract/EIP20Gateway');
+const SpyAssert = require('../../test_utils/SpyAssert');
+
+const { assert } = chai;
+
+describe('EIP20CoGateway.getEIP20BaseToken()', () => {
+  let web3;
+  let gatewayAddress;
+  let gateway;
+
+  let getBaseTokenAddressResult;
+
+  let spyCall;
+  let spyGetBaseTokenAddress;
+
+  const setup = () => {
+    spyGetBaseTokenAddress = sinon.replace(
+      gateway,
+      'getBaseToken',
+      sinon.fake.resolves(getBaseTokenAddressResult),
+    );
+    spyCall = sinon.spy(gateway, 'getEIP20BaseToken');
```

Comment:
I don't quite understand 🤔 
Mocks should be used to mock dependencies in unit tests so that your test only concerns the unit under test.
However, here it looks like you are mocking the unit under test.

If you mock the Gateway's methods, of course it will return the correct values. Then you are testing the mocking library.

Could this be the case in other tests as well?
But I am new to sinon, so I may be misreading this.

/cc @hobofan @sarvesh-ost

Reply:
I am not `mocking` the unit under test. I am `spying` it to check if the function is called with correct params.

Reply:
I was referring to this:
```js
    spyGetBaseTokenAddress = sinon.replace(
      gateway,
      'getBaseToken',
      sinon.fake.resolves(getBaseTokenAddressResult),
    );
```

Reply:
This looks correct to me.
The function under test is `getEIP20BaseToken`
and this line is mocking other function `getBaseToken`.

@hobofan please provide your feedback.